### PR TITLE
[WebNN] Support int4 and uint4 data types

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/model_builder.cc [hunk 1]
```diff
@@ -137,8 +137,16 @@ Status ModelBuilder::RegisterInitializers() {
           ORT_RETURN_IF_ERROR(onnxruntime::utils::UnpackInitializerData(tensor, unpacked_tensor));
           tensor_ptr = reinterpret_cast<std::byte*>(unpacked_tensor.data());
         }
+        if (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT4 ||
+            data_type == ONNX_NAMESPACE::TensorProto_DataType_UINT4) {
+          // For WebNN int4 and uint4 tensors are stored in Uint8Array,
+          // so we need to adjust the number of elements.
+          num_elements = static_cast<size_t>(std::ceil(static_cast<double>(num_elements) / 2.0));
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/model_builder.cc:144]:
```
          num_elements = (static_cast<size_t>(num_elements) + 1) / 2;
```

**Review by fdwr — APPROVED** [review]
One suggestion, else LGTM.

File: onnxruntime/core/providers/webnn/builders/model_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/model_builder.cc]: Thanks @fdwr, done.

**Review by fdwr — APPROVED** [review]
👍

**Comment by fdwr** [comment]
Restarting failed CI's (Big Models).

**Pull request merged**